crates/router/src/db/gsm.rs

```diff
@@ -104,12 +116,26 @@ impl GsmInterface for Store {
         sub_flow: String,
         code: String,
         message: String,
-        data: storage::GatewayStatusMappingUpdate,
-    ) -> CustomResult<storage::GatewayStatusMap, errors::StorageError> {
+        data: hyperswitch_domain_models::gsm::GatewayStatusMappingUpdate,
+    ) -> CustomResult<hyperswitch_domain_models::gsm::GatewayStatusMap, errors::StorageError> {
         let conn = connection::pg_connection_write(self).await?;
-        storage::GatewayStatusMap::update(&conn, connector, flow, sub_flow, code, message, data)
-            .await
-            .map_err(|error| report!(errors::StorageError::from(error)))
+        let gsm_update_data = diesel_models::gsm::GatewayStatusMappingUpdate::try_from(data)
+            .change_context(errors::StorageError::EncryptionError)?;
```

Comment:
This won't be encryption error, have a different variant for this error like invalid_data.